src/modules/configurator/hooks/useConfigEdit.tsx

```diff
@@ -5,44 +5,67 @@ import {
   dropdownOpen,
   dropdownState,
 } from "../state";
-import { useRecoilState, useSetRecoilState, useRecoilValue } from "recoil";
-import { fetchCarConfigurations } from "../services";
-import { useParams } from "react-router-dom";
+import { useSetRecoilState, useRecoilValue } from "recoil";
+import { fetchCarConfigurations, fetchUserConfiguration } from "../services";
+import { useSearchParams } from "react-router-dom";
 import { Timestamp } from "firebase/firestore";
+import { userState } from "../../authentification/state";
 
 export function useConfigEdit() {
-  const [configurations, setConfigurations] = useRecoilState(
-    currentConfigurationsState
-  );
+  const setConfigurations = useSetRecoilState(currentConfigurationsState);
   const setCurrentUserConfiguration = useSetRecoilState(userConfigurationState);
   const isDropdownOpen = useRecoilValue(dropdownOpen);
   const activeDropdownName = useRecoilValue(dropdownState);
+  const user = useRecoilValue(userState);
 
-  const { id } = useParams();
+  const [searchParams] = useSearchParams();
 
-  const handleCarConfigurationsFetch = async (id: string) => {
+  const modelId = searchParams.get("modelId");
+  const configId = searchParams.get("configId");
+
+  const handleCarConfigurationsFetch = async (modelId: string) => {
     try {
-      const response = await fetchCarConfigurations(id);
+      const defaultConfigurations = await fetchCarConfigurations(modelId);
+      setConfigurations(defaultConfigurations);
+
+      if (!user) throw new Error("User could not be found.");
```

Comment:
Should this maybe be triggered first?

Reply:
Default configurations are always needed because they get all properties of a car that can be used in editing. User id is only needed if a pre-existing user configuration is being edited to set the current active properties of user's configuration in the editor.